--- target
+++ candidate
@@ -24,7 +24,7 @@
 lea edi, dword [eax+0x13]
 jmp L84
 test al, 0x60
-je L149
+je L145
 call ADDR
 mov dword [esp+0x8], eax
 fild dword [esp+0x8]
@@ -33,10 +33,17 @@
 mov dword [esi+0x1cc], eax
 lea edi, dword [eax+0x16]
 cmp edi, -0x1
-je L149
+je L145
 mov al, byte [esi+0x41d]
 test al, al
-je L13e
+jne La8
+push edi
+mov ecx, ADDR
+call ADDR
+pop edi
+pop esi
+add esp, 0x1c
+ret
 mov eax, dword [ADDR]
 fld dword [eax+0x18c]
 fsub dword [esi+0x68]
@@ -61,14 +68,14 @@
 fcomp dword [ADDR]
 fnstsw ax
 test ah, 0x1
-je L10f
+je L11c
 mov dword [esp+0x8], 0x0
-jmp L128
+jmp L135
 fld dword [esp+0x8]
 fcomp dword [ADDR]
 fnstsw ax
 test ah, 0x41
-jne L128
+jne L135
 mov dword [esp+0x8], 0x3f800000
 mov eax, dword [esp+0x8]
 mov ecx, ADDR
@@ -79,10 +86,3 @@
 pop esi
 add esp, 0x1c
 ret
-push edi
-mov ecx, ADDR
-call ADDR
-pop edi
-pop esi
-add esp, 0x1c
-ret
